CI-Tests : IOZone Test Suite For NFS-Ganesha Client

mpadariy opened the pull request:
Added client script to run IOZone Filesystem Benchmark
suite on v3 , v4 and v4.1 mount for NFS-Ganehsa Client.

Signed-off-by: Mohammed Azhar Padariyakam <mpadariy@redhat.com>

mpadariy committed:
CI-Tests : IOZone Test Suite For NFS-Ganesha Client

Added client script to run IOZone Filesystem Benchmark
suite on v3 , v4 and v4.1 mount for NFS-Ganehsa Client.

Signed-off-by: Mohammed Azhar Padariyakam <mpadariy@redhat.com>

mpadariy committed:
CI-Tests : IOZone Test Suite For NFS-Ganesha Client

Added client script to run IOZone Filesystem Benchmark
suite on v3 , v4 and v4.1 mount for NFS-Ganehsa Client.

Signed-off-by: Mohammed Azhar Padariyakam <mpadariy@redhat.com>

dang commented on iozone/client.sh:
```diff
@@ -0,0 +1,95 @@
+
+if [ $ret -eq 0 ]
+then
+        echo "IOZone Test Is Completed And Successful";
```
Can the success lines also say which NFS version is in use?

mpadariy replied:
@dang Ok...I will fix it .. :)

dang commented on iozone/client.sh:
```diff
@@ -0,0 +1,95 @@
+
+umount -l /mnt/ganesha
+
+mount -t nfs -o vers=4 ${SERVER}:${EXPORT} /mnt/ganesha
```
This should probably explicitly say vers=4.0 since which minor version is tried first varies by kernel version.

mpadariy replied:
@dang sure...got it ... will change it to `4.0` frm `4`

dang commented on iozone/client.sh:
```diff
@@ -0,0 +1,95 @@
+        echo "IOZone Test Failed On NFSv4.1...";
+        tail -3 ../ioZoneLog.txt;
+        exit $ret;
+fi
```
Unmount at the end, just for completion?

mpadariy replied:
@dang Yea ... will fix that too .. :)

dang reviewed (commented):
Looks pretty good.  A few minor nits.

mpadariy committed:
Merge branch 'ioZoneTestSuite' of https://github.com/mpadariy/ci-tests into ioZoneTestSuite

mpadariy commented:
@dang Hi ... :)
I have fixed all the issues and raised the PR .. :)

soumyakoduri commented on iozone/client.sh:
```diff
@@ -0,0 +1,98 @@
+mkdir -p /mnt/ganesha
+
+cd ~
+curl -o iozone3_394.tar http://www.iozone.org/src/current/iozone3_394.tar
```
nit: If possible its better to get a latest stable iozone binary available instead of sticking to iozone3_394 version.

mpadariy replied:
@soumyakoduri Sure ... will look into it .. :)

soumyakoduri replied:
Thanks! This can be enhanced later (as part of another pull request).

nixpanic replied:
We could put the iozone RPM in the centos-gluster*-test repository. That way you can install it with

    # yum --enablerepo=centos-gluster*test -y install iozone

mpadariy replied:
@nixpanic 
Hi, How can I add the iozone RPM to `centos-gluster*test` ?

malahal commented:
Can't we simulate iozone workload with fio?

On Tue, Nov 7, 2017 at 5:34 PM, Niels de Vos <notifications@github.com>
wrote:

> *@nixpanic* commented on this pull request.
> ------------------------------
>
> In iozone/client.sh
> <https://github.com/nfs-ganesha/ci-tests/pull/19#discussion_r149347103>:
>
> > +#  - SERVER: hostname or IP-address of the NFS-server
> +#  - EXPORT: NFS-export to test (should start with "/")
> +
> +# enable some more output
> +set -x
> +
> +[ -n "${SERVER}" ]
> +[ -n "${EXPORT}" ]
> +
> +# install build and runtime dependencies
> +yum -y install git gcc nfs-utils time
> +
> +mkdir -p /mnt/ganesha
> +
> +cd ~
> +curl -o iozone3_394.tar http://www.iozone.org/src/current/iozone3_394.tar
>
> We could put the iozone RPM in the centos-gluster*-test repository. That
> way you can install it with
>
> # yum --enablerepo=centos-gluster*test -y install iozone
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nfs-ganesha/ci-tests/pull/19#discussion_r149347103>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAxtbGRaunIXvFoL1TUeK5aAwqFj3OLLks5s0EdGgaJpZM4QHZzJ>
> .
>

mpadariy committed:
Update client.sh

Changed curl iozone to yum install iozone from centos-gluster*-test

mpadariy committed:
Update client.sh

Mods to yum install to incorporate centos-gluster-test repo

nixpanic commented on iozone/client.sh:
```diff
@@ -0,0 +1,103 @@
+#cd iozone3_394/src/current
+#make
+#make linux
+#ioZoneDir=$(pwd)/iozone
```
Could you remove these leftovers, please? There are some more further below, the #ioZoneDir commands.

mpadariy replied:
@nixpanic Sure .. :)

mpadariy replied:
@nixpanic All comments are removed from the script ... :) 
Console output of final build using this script ..
https://ci.centos.org/view/NFS-Ganesha/job/Azhar_TestJob/3/console

mpadariy committed:
Update client.sh

Finalized client script for Iozone test suite on nfs-ganesha v3,v4.0,v4.1 mounts

nixpanic commented on iozone/client.sh:
```diff
@@ -0,0 +1,91 @@
+        exit $ret;
+fi
+
+umount -l /mnt/ganesha
```
Is there a reason to use a lazy unmount? In case unmounting fails, that will now not get caught (and it could be a nfs-ganesha problem). In normal circumstances unmounting should always be successful.

mpadariy replied:
@nixpanic 
There wasnt any particular reason...But as you pointed out ... when I was working with gluster volumes...umount failed sometimes ... and when I used the lazy approach ... umount was successful...But that really hides the server/client issues during the umount procedures ... 
That should be rectified .. :)

mpadariy committed:
Changed umount strategy

Updated umount startegy from lazy umount to normal. Since lazy umount may bypass any server issues that results during umount.

mpadariy commented:
@nixpanic Changed the umount method ... :)

mpadariy committed:
Update client.sh

nixpanic commented:
@mpadariy could you post a link with the console output of a test-run with the latest version of the scripts?

mpadariy commented:
@nixpanic Hi .. The Tests are Successful :) 
Console Output : 
https://ci.centos.org/view/NFS-Ganesha/job/Azhar_TestJob/4/console
